examples/rust/regex/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+wit_bindgen_rust::export!("s2regex.wit");
+use regex::Regex;
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+struct S2regex;
+
+thread_local! {
+    static COMPILED_RGXS: RefCell<HashMap<String, Regex>> = RefCell::new(HashMap::new());
+}
+
+impl s2regex::S2regex for S2regex {
+    fn capture(input: String, pattern: String) -> String {
+        if !COMPILED_RGXS.with(|c| c.borrow().contains_key(&input)) {
+            let re = Regex::new(pattern.as_str()).unwrap();
+            COMPILED_RGXS.with(|c| c.borrow_mut().insert(pattern.clone(), re));
+        }
+        COMPILED_RGXS.with(|c| {
```

Comment:
just curious, but couldn't we merge these three borrows into a single borrow_mut? Is there an advantage to keeping them separate?

Reply:
Yeah, this could probably be more elegantly expressed.  We could at least probably refactor out those 3 `COMPILED_RGXS.with` parts.  @sadroeck Would you mind cleaning up my mess here a bit?  ;-)

Reply:
Collapsed it down into a single borrow & simplified the get-or-insert pattern a little in 192e604.
Should've done this the first time around :s